python/src/aiconfig/editor/client/src/components/ParametersRenderer.tsx

```diff
@@ -0,0 +1,254 @@
+          radius="md"
+          size="xs"
+          value={parameterName}
+          onChange={(event) => {
+            setParameterName(event.target.value);
+            if (event.target.value) {
+              debouncedCellParameterUpdate(
+                event.target.value,
+                parameterValueString
+              );
+            }
+          }}
+        />
+        <Textarea
+          placeholder="Enter parameter value"
+          disabled={isReadonly}
+          radius="md"
+          value={parameterValueString}
+          autosize
+          size="xs"
+          maxRows={5}
+          onChange={(event) => {
+            setParameterValueString(event.target.value);
+            debouncedCellParameterUpdate(parameterName, event.target.value);
+          }}
+        />
+      </Stack>
+      <ActionIcon
+        onClick={() => removeParameter(parameterName)}
+        style={{ marginTop: -50 }}
+        disabled={isReadonly}
+      >
+        <IconTrash size={16} color={isReadonly ? "grey" : "red"} />
+      </ActionIcon>
+    </Group>
+  );
+});
+
+export type ParametersArray = {
+  parameterName: string;
+  parameterValue: JSONValue;
+  key: string;
+}[];
+
+export default memo(function ParametersRenderer(props: {
+  initialValue?: JSONObject;
+  onUpdateParameters: (data: {
+    promptName?: string;
+    newParameters: ParametersArray;
+  }) => void;
+  customDescription?: React.ReactNode;
+  maxHeight?: string | number;
+}) {
+  const { initialValue, onUpdateParameters } = props;
+  // TODO: saqadri - update this when we have a readonly mode
+  const { isReadonly } = { isReadonly: false }; //useContext(WorkbookContext);
+
+  const [parameters, setParameters] = useState<ParametersArray>(
+    initialValue && Object.keys(initialValue).length > 0
+      ? Object.keys(initialValue).map((parameterName) => {
+          return {
+            key: parameterName,
+            parameterName,
+            parameterValue: initialValue[parameterName],
+          };
+        })
+      : [
+          {
+            key: uniqueId(),
+            parameterName: "",
+            parameterValue: "",
+          },
+        ]
+  );
+
+  const removeParameter = useCallback(
+    async (key: string, parameterName?: string) => {
+      setParameters((prev) => {
+        const newParameters = prev.filter((item) => item.key !== key);
+        onUpdateParameters({ newParameters });
+        return newParameters;
+      });
+    },
+    [setParameters, onUpdateParameters]
+  );
+
+  const addParameter = useCallback(async () => {
+    setParameters((prev) => {
+      const newParameters = [
+        ...prev,
+        {
+          key: uniqueId(),
+          parameterName: "",
+          parameterValue: "",
+        },
+      ];
+      onUpdateParameters({ newParameters });
+      return newParameters;
+    });
+  }, [onUpdateParameters]);
+
+  // TODO: saqadri - add MantineProvider wrapper inside EditorContainer in order for this to pick up context
```

Comment:
nit, probably inside Editor. The Editor component should be completely interchangeable between clients (e.g. here, gradio, etc.) with EditorContainer ideally completely reusable with some configuration via callbacks